### PR TITLE
Di, cleanup

### PR DESCRIPTION
**Pull request by Yahasana**

### CODE REVIEW

File: ice/di/access.zep [hunk 1]
```diff
@@ -25,15 +25,13 @@ class Access
     {
         var di, service;
 
-        if typeof this->di != "object" || typeof this->di == "object" && !(this->di instanceof Di) {
```
Comment (mruz) [inline ice/di/access.zep:28]: I remember it wasn't possible to check `instanceof` on non-object on zephir before

Comment (Yahasana) [reply ice/di/access.zep:28]: alright, updated

**Comment by mruz** [comment]
Could you please add a simple test, please? \Ice\Mcv\Service extends Access, so

`tests/App/Services/UserService.php`
```php
<?php

namespace Tests\App\Services;

use Ice\Mvc\Service;
use Tests\App\Models\Users;

class UserService extends Service
{
    public function __construct(Users $model)
    {
        $this->setModel($model);
    }
}
```

and to `tests/ModelsTest.php`
```php
    /**
     * Service test
     */
    public function testService()
    {
        $service = new \Tests\App\Services\UserService(new Users());

        $this->assertInstanceOf(\Ice\Di::class, $service->di);
        $this->assertInstanceOf(\Ice\Config::class, $service->config);
        $this->assertInstanceOf(\Ice\Db::class, $service->db);
    }
```

then we could try the previous version, maybe it works with the latest Zephir.

**Comment by Yahasana** [comment]
i have updated and just removed redendent is object test. No any other changes

**Pull request merged**